cluster.tf

```diff
@@ -32,10 +32,18 @@ resource "aws_eks_cluster" "this" {
     aws_iam_role_policy_attachment.cluster_AmazonEKSServicePolicy,
     aws_cloudwatch_log_group.this
   ]
+}
+
+resource "null_resource" "wait_for_cluster" {
+  depends_on = [
+    aws_eks_cluster.this[0]
+  ]
+
   provisioner "local-exec" {
-    command = <<EOT
-    until curl -k -s ${aws_eks_cluster.this[0].endpoint}/healthz >/dev/null; do sleep 4; done
-  EOT
+    command = var.manage_aws_auth ? var.wait_for_cluster_cmd : ""
+    environment = {
+      ENDPOINT = aws_eks_cluster.this[0].endpoint
```

Comment:
I believe this is close to what you were expressing? Not sure :) but it does seem to work.

Reply:
Yes. This is much cleaner IMO 😃